processor/src/host/mod.rs

```diff
@@ -75,23 +75,25 @@ pub trait BaseHost {
         &mut self,
         process: &mut ProcessState,
         options: &DebugOptions,
-    ) -> Result<(), ExecutionError> {
+    ) -> Result<(), DebugError> {
```

Comment:
I think the canonical way to do this would be to make those errors associated types of `BaseHost`:

```rs
pub trait BaseHost {
  type DebugError: Error + Send + Sync + 'static;
  type AssertError: Error + Send + Sync + 'static;
  type TraitError: Error + Send + Sync + 'static;

  ...
}
```